### PR TITLE
#40/new tab

### LINKED ISSUE #66: Popup window size is defined by the current browser window size
### Description of the Bug

Popups will be full screen if the browser is full screen and half of the screen only if the browser window is on a split

### What should happen instead

Popups should always be generated with the dimensions indicated in the event details, regardless of the current size of the browser window

### Steps to reproduce this bug

click the 'trigger popups' button or wait out the count down on the next alarm 

### What could we do to solve this bug?

_No response_

### Assignment

This issue is free for anyone to take

### PR DESCRIPTION
**Pull request by nichgalzin**
# Description

This is accompanied by a minor backed pr [here](https://github.com/enBloc-org/arebyte-plugin-backend/pull/24)

**Relates #40 & Closes #66**  
This PR works primarily on the background processing of popup data into a streamlined form that can be stored into storage and then accessed from the frontend of each popup window. It includes several new utils and their accompanying tests. 

#### New utils
- determineFormate.js - looks at the extension of media-content components and determines if they are a support image or video format.
- parseWindowSize.js - provides the logic for setting window size and aspect ratio for popup windows based on user values 'large', 'medium' etc. (used for text and video content)
- createWindow.ts - function that is responsible for creating popup window and parameters

### UI changes

Screenshot 2024-10-24 at 10 35 04

# Tests

Include a description of any tests added - if none have been added explain why.

- src/__tests__/determineFormate.test.js
- src/__tests__/parseImageSize.test.js
- src/__tests__/parseImageSize.test.js

### CODE REVIEW

File: src/__tests__/determineFormate.test.js
Comment (camelPhonso) [inline src/__tests__/determineFormate.test.js:1]: I'm surprised this typo didn't stop the test from running in the ci, but it's bound to give us some headaches I imagine

File: src/tabs/newTab.tsx [hunk 1]
```diff
@@ -1,16 +1,67 @@
...
+        </>
+      )}
+
+      {/* <img
         src={url}
         alt="an image"
-        style={{ width: Number(width), height: Number(height) }}
-      />
+        style={{ width: width, height: height }}
+      /> */}
```
Comment (camelPhonso) [inline src/tabs/newTab.tsx]: I suppose we don't want this here

File: src/tabs/newTab.tsx [hunk 2]
```diff
@@ -1,16 +1,67 @@
...
   const params = new URLSearchParams(window.location.search)
-  const width = params.get("width")
-  const height = params.get("height")
-  const url = "http://localhost:1337" + params.get("url")
-
+  const index = params.get("index")
+  const [popup, setPopup] = useState<SlimPopup>()
+  useEffect(() => {
+    const getFromStorage = async () => {
+      const { popups } = await Browser.storage.session.get(["popups"])
```
Comment (camelPhonso) [inline src/tabs/newTab.tsx]: this could not get noticed for a long time but just as a general best practice we should name keys going into storage in a more 'unique' way. "popup" is fairly generic and we run the risk of our storage being overridden.

Again, it's a fairly reduced risk because of how we're using this in a single session and all that, but just one of those that could lead to a super edge issue that would take for ever to uncover if it happened

File: src/tabs/newTab.tsx [hunk 3]
```diff
@@ -1,16 +1,67 @@
...
+              <h1>{popup.popupInfo.work_title}</h1>
+              <BlocksRenderer content={popup.text_content} />
+            </>
+          )}
+          {popup.type === "image" && (
+            <img
+              src={
+                process.env.NODE_ENV === "development"
+                  ? "http://localhost:1337" + popup.url
+                  : popup.url
```
Comment (camelPhonso) [inline src/tabs/newTab.tsx:40]: this is because of that issue where strapi manipulates the url returned differently between `dev` and `production`?

**Review by camelPhonso — CHANGES_REQUESTED** [review]
It's looking great. The tests and documentation are sweet xD

Just some fiddly things with file names and whatnot

File: src/__tests__/determineFormate.test.js
Comment (nichgalzin) [reply src/__tests__/determineFormate.test.js:1]: Fixed. Good catch, doesn't stop anything from running though. Can name the fill anything we want no? They don't need to be called anything particular. Just need to have the .test. format

File: src/tabs/newTab.tsx (hunk 3 above)
Comment (nichgalzin) [reply src/tabs/newTab.tsx:40]: Yes exactly. When the url will come from the production deployed cms it will be a complete url to the stored image url in the bucket. Mostly put this here as reminder to myself

File: src/tabs/newTab.tsx (hunk 1 above)
Comment (nichgalzin) [reply src/tabs/newTab.tsx]: okeeeee

File: src/tabs/newTab.tsx (hunk 2 above)
Comment (nichgalzin) [reply src/tabs/newTab.tsx]: This is a good call.

**Review by camelPhonso — APPROVED** [review]
nice one 👌
